### PR TITLE
Implement remote configuration rules toggling with ASM product

### PR DESCRIPTION
**Pull request by CarlesDD**
### What does this PR do?
Implement remote configuration rules toggling with ASM product

### Motivation
Remote configuration ASM product allows to toggle rules in ASM.

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/appsec/remote_config/index.js [hunk 1]
```diff
@@ -32,6 +32,18 @@ function enable (config) {
   }
 }
 
+function enableAsm () {
+  if (rc) {
+    rc.on('ASM', RuleManager.toggleRules)
```
Comment (uurien) [inline packages/dd-trace/src/appsec/remote_config/index.js:37]: we should enable `ASM_DD_RULES` capability

File: packages/dd-trace/src/appsec/remote_config/index.js [hunk 2]
```diff
@@ -32,6 +32,18 @@ function enable (config) {
...
+  }
+}
+
+function disableAsm () {
+  if (rc) {
+    rc.off('ASM', RuleManager.toggleRules)
```
Comment (uurien) [inline packages/dd-trace/src/appsec/remote_config/index.js:43]: and disable the capability here

File: packages/dd-trace/src/appsec/remote_config/index.js [hunk 3]
```diff
@@ -32,6 +32,18 @@ function enable (config) {
   }
 }
 
+function enableAsm () {
+  if (rc) {
```
Comment (uurien) [inline packages/dd-trace/src/appsec/remote_config/index.js:36]: if rules are set by customer, we shouldn't enable the capability neither subscribe to `ASM`

File: packages/dd-trace/src/appsec/rule_manager.js [hunk 4]
```diff
@@ -69,6 +69,23 @@ function copyRulesData (rulesData) {
...
+
+function getRulesOverrideData (asmData) {
+  return asmData.rules_override && Array.isArray(asmData.rules_override) &&
+    asmData.rules_override
+      .filter(ruleOverride => ruleOverride.hasOwnProperty('enabled'))
+      .map(ruleOverride => ({ id: ruleOverride.id, enabled: ruleOverride.enabled }))
+}
+
+function toggleRules (action, asmData) {
+  const rulesOverrideData = getRulesOverrideData(asmData)
```
Comment (uurien) [inline packages/dd-trace/src/appsec/rule_manager.js:81]: What happens when the rule is disabled and enabled again from backend? we receive the rule with `enabled: true` again?

**Comment by CarlesDD** [comment]
Closing in favour of [Update appsec rules from Remote Config](https://github.com/DataDog/dd-trace-js/pull/2654) which includes the feat with the new WAF spec.

**Pull request closed without merging**